src/Features/CSharp/Portable/UseDefaultExpression/CSharpUseDefaultExpressionCodeFixProvider.cs

```diff
@@ -0,0 +1,89 @@
+        private const string CS8107 = nameof(CS8107); // Feature is not available in C# 7.0. Please use language version X or greater.
+        private const string CS8059 = nameof(CS8059); // Feature is not available in C# 6. Please use language version X or greater.
+        private const string CS8026 = nameof(CS8026); // Feature is not available in C# 5. Please use language version X or greater.
+        private const string CS8025 = nameof(CS8025); // Feature is not available in C# 4. Please use language version X or greater.
+        private const string CS8024 = nameof(CS8024); // Feature is not available in C# 3. Please use language version X or greater.
+        private const string CS8023 = nameof(CS8023); // Feature is not available in C# 2. Please use language version X or greater.
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(CS8313, CS8363, CS8107, CS8059, CS8026, CS8025, CS8024, CS8023);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // This code fix addresses very specific compiler errors. It's unlikely there will be more than 1 of them at a time.
+            return null;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var token = syntaxRoot.FindToken(context.Span.Start);
+
+            if (token.Span == context.Span &&
+                token.IsKind(SyntaxKind.DefaultKeyword) &&
+                token.Parent.IsKind(SyntaxKind.DefaultLiteralExpression))
+            {
+                // If there happens to be more than 1 diagnostic (for example a default literal in a case label in C# 7.0),
+                // we will fix all of them, so pass in context.Diagnostics, not just the first one.
+                context.RegisterCodeFix(new MyCodeAction(
+                    c => FixAsync(context.Document, context.Span, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        private static async Task<Document> FixAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var defaultToken = syntaxRoot.FindToken(span.Start);
+            var defaultLiteral = (LiteralExpressionSyntax)defaultToken.Parent;
+
+            var type = semanticModel.GetTypeInfo(defaultLiteral, cancellationToken).ConvertedType;
+            if (type.IsAnonymousType)
```

Comment:
what if ConvertedType is null?

Reply:
It was never null in the tests, which cover cases like a missing expression, var, overload resolution failure, an anonymous type, a lambda expression. Do you know if it's possible for this to be null? And if so, when would that happen?

Reply:
Huh, in C# 7.1 it is indeed null, but in 7.0, it's an error type. We only offer this inside expressions on 7.0 and lower, that's why the tests all passed.

Reply:
➡️ Null check added in 20d75193b6468f52745bb3d901d5759743557c87